src/common/transport/transport-client.js

```diff
@@ -0,0 +1,80 @@
+import { TransportClientError } from '../errors';
+import rp from 'request-promise';
```

Comment:
I agree with dropping bluebird as soon as possible. We should use as much ES6 as possible. 

I think more people are familiar with request but I think the ability to run outside of nodejs is more important.